synapse/util/iterutils.py

```diff
@@ -71,6 +73,31 @@ def chunk_seq(iseq: S, maxlen: int) -> Iterator[S]:
     return (iseq[i : i + maxlen] for i in range(0, len(iseq), maxlen))
 
 
+def partition(
+    iterable: Iterable[T], predicate: Callable[[T], bool]
```

Comment:
`itertools` seems to use argument order as `partition(predicate, iterable)`. Should we just standardize with them to avoid surprises for people used to that?

My inclination would be to have it `partition(iterable, predicate)` this way and is what @clokep used in some off-hand pseudo-code.

Reply:
I think I have a slight preference for the iterabile first since it might get lost in the lambda otherwise? Definitely not a strong preference though. Type checking should mostly save us from the silly error.